src/base.iced

```diff
@@ -514,9 +521,23 @@ class Base
     #   SEMIPRIVATE : 3
     #
     ret.seq_type = @seq_type if @seq_type?
-
     ret.ignore_if_unsupported = !!@ignore_if_unsupported if @ignore_if_unsupported?
 
+    if @hprev_hash? and not @hprev_seqno?
```

Comment:
We don't seem to be doing any sanity checking in this function. I wonder if it's happening somewhere else on the *generate* path (not the *verify* path).

Reply:
If we do decide to keep these checks, please do something like:

```iced
await @_sanity_check_hprev { @hprev_hash, @hprev_seqno }, esc defer()
```

i.e, try to match the style of the existing code.